src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -151,50 +150,31 @@ ConcurrencyWorker::PrepAndSendInferRequests()
   // Non-sequence model is 'num_reqs' * 1 ctx
   // Sequence model is 1 request of 1 sequence * 'active_ctx_cnt' ctxs_
   while (total_ongoing_requests_ < (int)thread_config_->concurrency_ &&
-         early_exit == false) {
-    PrepAndSendInferRequest();
+         early_exit == false && execute_ && thread_stat_->status_.IsOk()) {
```

Comment:
Why was `thread_stat_->status_.IsOk()` added here?

Reply:
Same reason as https://github.com/triton-inference-server/client/pull/226#discussion_r1081618917
Old code would sometimes exit the loop so we could evaluate [HandleExitConditions()](https://github.com/triton-inference-server/client/blob/refactor-managers/src/c%2B%2B/perf_analyzer/concurrency_worker.cc#L68) and [this exit condition](https://github.com/triton-inference-server/client/blob/refactor-managers/src/c%2B%2B/perf_analyzer/concurrency_worker.cc#L68).